tests/test_function_utils.py

```diff
@@ -32,3 +33,50 @@ def boo(param_1, param_2):
         key6 = build_value_by_func_parameters('xyz_{param_1}_{param_2}', boo, None, None)
         self.assertNotEqual(key6, None)
         self.assertEqual(key6, 'xyz_None_None')
+
+    def test_func_param_to_dict(self):
+        def boo(param_1, param_2):
+            return 1
+
+        dict1 = get_func_parameters_as_dict(boo, 1, 2)
+        self.assertNotEqual(dict1, None)
+        self.assertEqual(dict1['param_1'], 1)
+        self.assertEqual(dict1['param_2'], 2)
+
+        dict2 = get_func_parameters_as_dict(boo, *[1, 2], **{})
+        self.assertNotEqual(dict2, None)
+        self.assertEqual(dict2['param_1'], 1)
+        self.assertEqual(dict2['param_2'], 2)
+
+        dict3 = get_func_parameters_as_dict(boo, *[11], **{'param_2': 22})
+        self.assertNotEqual(dict3, None)
+        self.assertEqual(dict3['param_1'], 11)
+        self.assertEqual(dict3['param_2'], 22)
+
+        dict4 = get_func_parameters_as_dict(boo, None, None)
+        self.assertNotEqual(dict4, None)
+        self.assertEqual(dict4['param_1'], None)
+        self.assertEqual(dict4['param_2'], None)
+
+    def test_func_param_index_by_name(self):
+        def boo(param_1, param_2, param_3):
+            return 1
+
+        def foo(test_param):
+            return 1
+
+        index1 = get_func_parameter_index_by_name(boo, "param_1")
+        self.assertNotEqual(index1, None)
```

Comment:
if it will be equal to 0 in the next assert, does this mean it's not None? 
do we need the None assertion?

None assertions are good in cases when functions the return objects. 
before accessing the object properties we will check the object is Not None. 
for the reason that the test will not crush